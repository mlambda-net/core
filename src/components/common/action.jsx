import * as React from 'react';

class Action extends React.Component {
  constructor(props) {
    super(props);
    const { children } = props;
    this.state = {
      routes: React.Children.map(children, (child) => {
        return {
          name: child.props.name,
          component: child.props.children,
        };
      }),
    };
  }

  component() {
    const { routes } = this.state;
    const items = routes.filter((r) => r.name === this.props.actual);
    if (items.length === 0) {
      return <span />;
    }
    return items[0].component;
  }

  render() {
    return this.component();
  }
}

export default Action;
