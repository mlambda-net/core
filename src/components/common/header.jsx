import React from 'react';
import { AppBar, IconButton, Toolbar, Typography } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = () => ({
  root: {
    flexGrow: 1,
  },
  title: {
    flexGrow: 1,
  },
});

class Header extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { children, name, classes } = this.props;
    return (
      <div className={classes.root}>
        <AppBar position="fixed" className={this.props.className}>
          <Toolbar>
            <IconButton
              style={{ display: this.props.open ? 'none' : '' }}
              edge="start"
              color="inherit"
              aria-label="menu"
              onClick={this.props.onClick}
              size="large">
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" className={classes.title}>
              {name}
            </Typography>
            {children}
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

export default withThemes(styles)(Header);
