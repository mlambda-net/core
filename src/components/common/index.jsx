export { default as Label } from './label';
export { default as LeftImage } from './leftimage';
export { default as Option } from './option';
export { default as Title } from './title';
export { default as Menu } from './menu';
export { default as Navigation } from './navigation';
export { default as Header } from './header';
export { default as withValidation } from './validation';
export { default as Action } from './action';
export { default as State } from './state';
