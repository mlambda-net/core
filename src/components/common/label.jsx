import React from 'react';
import { Box, Typography } from '@mui/material';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = () => ({
  label: {
    fontSize: '0.9em',
  },
  value: {
    fontSize: '0.8em',
  },
});

class Label extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { classes } = this.props;

    return (
      <Box style={this.props.style} className={this.props.className}>
        <Typography color="textPrimary" className={classes.label}>
          {this.props.label}
        </Typography>
        <Typography color="textSecondary" className={classes.value}>
          {this.props.value}
        </Typography>
      </Box>
    );
  }
}

export default withThemes(styles)(Label);
