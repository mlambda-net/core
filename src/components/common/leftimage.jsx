import React from 'react';
import { Box, Grid, Paper } from '@mui/material';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = () => ({});

class LeftImage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Paper elevation={10} className={this.props.className}>
        <Box p={3}>
          <Grid container spacing={0}>
            <Grid item xs={'auto'} sm={6} height="100%">
              <Box
                height="100%"
                display="flex"
                justifyContent="center"
                alignItems="center">
                {this.props.img}
              </Box>
            </Grid>
            <Grid item xs={12} sm={6}>
              <Box
                height="100%"
                display="flex"
                justifyContent="center"
                alignItems="center">
                {this.props.children}
              </Box>
            </Grid>
          </Grid>
        </Box>
      </Paper>
    );
  }
}

export default withThemes(styles)(LeftImage);
