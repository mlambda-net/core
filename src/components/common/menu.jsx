import React from 'react';
import { Button } from '@mui/material';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = (theme) => ({
  menu: {
    border: '3px solid',
    borderColor: theme.palette.text.primary,
    textAlign: 'center',
    fontSize: theme.typography.subtitle1.fontSize,
    fontFamily: theme.typography.fontFamily,
    color: theme.palette.text.primary,
    cursor: 'pointer',
  },
});

class Menu extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <Button variant="outlined" className={classes.menu}>
        {this.props.name}
      </Button>
    );
  }
}

export default withThemes(styles)(Menu);
