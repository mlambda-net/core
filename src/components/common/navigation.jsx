import React from 'react';
import { Divider, Drawer, IconButton, Typography } from '@mui/material';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = (theme) => ({
  root: {
    display: 'flex',
  },
  drawer: {
    flexShrink: 0,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    ...theme.mixins.toolbar,
  },
  paper: {
    width: 'inherit',
  },
  title: {
    flexGrow: 1,
  },
});

class Navigation extends React.Component {
  constructor(props) {
    super(props);
    this.onChange = this.emitChange.bind(this);
  }

  emitChange(event) {
    this.props.onClose(event);
    event.preventDefault();
  }

  render = () => {
    const { name, position, classes, open, children } = this.props;
    return (
      <div className={classes.root}>
        <Drawer
          style={{ width: open ? this.props.width : '0' }}
          open={open}
          className={classes.drawer}
          variant="persistent"
          anchor={position}
          classes={{
            paper: classes.paper,
          }}>
          <div className={classes.drawerHeader}>
            <Typography variant="h6" className={classes.title}>
              {name}
            </Typography>
            <IconButton color="secondary" onClick={this.onChange} size="large">
              <ArrowBackIcon />
            </IconButton>
          </div>
          <Divider />
          {children}
        </Drawer>
      </div>
    );
  };
}

export default withThemes(styles)(Navigation);
