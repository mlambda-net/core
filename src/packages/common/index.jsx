export Label from './label';
export LeftImage from './leftimage';
export Option from './option';
export Title from './title';
export * from './validation';
export Menu from './menu';
export Navigation from './navigation';
export Header from './header';
