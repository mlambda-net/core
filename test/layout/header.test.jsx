import React from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from 'vitest';
import { Header } from '@mlambda-net/web-core/common';
import Wrap from '../util';

test('Header render', () => {
  const item = renderer.create(
    <Wrap>
      <Header name="app">
        <div>hello</div>
      </Header>
    </Wrap>,
  );

  let tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});
