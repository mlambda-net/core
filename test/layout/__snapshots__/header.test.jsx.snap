// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header render 1`] = `
<div
  className="Component-root-1"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-17zs8v5-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-158p51c-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-17y397t-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "display": "",
          }
        }
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-havevq-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <h6
        className="MuiTypography-root MuiTypography-h6 Component-title-2 css-1gn1d81-MuiTypography-root"
      >
        app
      </h6>
      <div>
        hello
      </div>
    </div>
  </header>
</div>
`;
